### PR TITLE
Implemented `Box()` function and improved boxing

### LINKED ISSUE #7: Invalid check when on the left side does not work (boxing issue)
The code below exemplifies the issue, when `invalid` is used on the left side of the test it is not boxed in `roInvalid` so is not correctly compared with the parameter (that was boxed because of the `object` type on the argument).

```brightscript
sub main()
  myVar = invalid
  RightInvalid(myVar)
  LeftInvalid(myVar)
end sub

sub RightInvalid(param1 as object)
  if param1 <> invalid
     param1.id = 1
  else
     print "param invalid"
  end if
end sub

sub LeftInvalid(param1 as object)
  if invalid <> param1
     param1.id = 1
  else
     print "param invalid"
  end if
end sub
```
The code was supposed to print `param invalid` in two consecutive lines, instead it shows the error below, because the comparison cannot say `invalid` is not different than `roInvalid`:
```shell
param invalid
left-invalid-error.brs(17,12-14): Attempting to set property on non-iterable value
    left: Object
```

### PR DESCRIPTION
**Pull request by lvcabral**
The boxed Numbers and Boolean were not being properly handled, was not possible to compare or make arithmetic operations among them nor with intrinsic values. Now is fixed.

This PR closes #7 and #18

### CODE REVIEW

File: src/interpreter/index.ts [hunk 1]
```diff
@@ -1350,13 +1357,13 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
...
         let target = this.evaluate(statement.target);
         if (!isIterable(target)) {
-            return this.addError(
-                new BrsError(
-                    `Attempting to iterate across values of non-iterable type ` +
-                        ValueKind.toString(target.kind),
-                    statement.item.location
-                )
-            );
+            // Roku device does not crash if the value is not iterable, just send a console message
```
Comment (TwitchBronBron) [inline src/interpreter/index.ts:1360]: Nice catch! Glad to see aligning with the actual runtime behavior.

File: src/interpreter/index.ts [hunk 2]
```diff
@@ -1650,6 +1663,40 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         return value;
     }
 
+    debugLocalVariables(): string {
```
Comment (TwitchBronBron) [inline src/interpreter/index.ts:1666]: Could you add a jsdoc comment explaining the purpose of this function? It feels like an internal-to-brs function but I'm not sure what it's for just looking at it.

File: test/e2e/BrsComponents.test.js [hunk 3]
```diff
@@ -414,15 +414,15 @@ describe("end to end brightscript functions", () => {
             "Comparing true = false should be false ",
             "false",
             "Double value ",
-            "123.456",
+            " 123.456",
```
Comment (TwitchBronBron) [inline test/e2e/BrsComponents.test.js:417]: More device behavior alignment fixes, nice!

**Review by TwitchBronBron — APPROVED** [review]

File: src/interpreter/index.ts (hunk 2 above)
Comment (lvcabral) [reply src/interpreter/index.ts:1666]: I will add, this is a method to show the Variables on the REPL interface (TTY) but also will be used when I integrate the MicroDebugger

**Pull request merged**